lib/ueberauth/strategy/strava.ex

```diff
@@ -13,22 +15,26 @@ defmodule Ueberauth.Strategy.Strava do
   Handles initial request for Strava authentication.
   """
   def handle_request!(conn) do
+    scopes = conn.params["scope"] || option(conn, :default_scope)
+    opts = [redirect_uri: callback_url(conn), scope: scopes]
 
-    authorize_url = conn.params
-      |> maybe_replace_param(conn, "scope", :default_scope)
-      |> Enum.map(fn {k,v} -> {String.to_existing_atom(k), v} end)
-      |> Keyword.put(:redirect_uri, callback_url(conn))
-      |> Ueberauth.Strategy.Strava.OAuth.authorize_url!
+    opts = if conn.params["state"] do 
+             Keyword.put(opts, :state, conn.params["state"])
```

Comment:
Again, just went with more recent strategies, I can revert if it doesn't suite taste